packages/integrations/mdx/src/index.ts

```diff
@@ -0,0 +1,38 @@
+import type { AstroIntegration } from 'astro';
+import mdxPlugin from '@mdx-js/rollup';
+
+export default function mdx(): AstroIntegration {
+	return {
+			name: '@astrojs/mdx',
+			hooks: {
+				'astro:config:setup': ({ updateConfig, addPageExtension, command }: any) => {
+					addPageExtension('.mdx');
+					updateConfig({
+						vite: {
+							plugins: [
+								{ 	
+									enforce: 'pre',
+									...mdxPlugin({
+										jsx: true,
+										jsxImportSource: 'astro',
+										// Note: disable `.md` support
+										format: 'mdx',
+										mdExtensions: []
+									})
+								},
+								command === 'dev' && {
+									name: '@astrojs/mdx',
+									transform(code: string, id: string) {
+										if (!id.endsWith('.mdx')) return;
+										return `${code}\nif (import.meta.hot) {
```

Comment:
All for disabling HMR before finding a stable solution. May be worth a TODO to circle back to?

Reply:
Added in [c33dbe4](https://github.com/withastro/astro/pull/3706/commits/c33dbe44a263cdb4e35e5161aa9df71ea71a0000)